### PR TITLE
ELPP-3665 Define surrogate keys for iiif

### PR DESCRIPTION
**Pull request by giorgiosironi**
Also refine them for generic-cdn. Will allow the bot to invalidate a single article or a single version.

### CODE REVIEW

File: projects/elife.yaml [hunk 1]
```diff
@@ -773,7 +773,7 @@ generic-cdn:
             surrogate-keys:
                 article-id:
                     url: "^/articles/([0-9]+)/(.+)$"
-                    value: "articles/\\1"
+                    value: "articles/\\1 article/\\1"
```
Comment (giorgiosironi) [inline projects/elife.yaml]: Will define `version/...` here too, but there are file like `/articles/10627/elife-10627-v1-download.pdf` that makes it a bit more convoluted. Are those still used?

File: projects/elife.yaml [hunk 2]
```diff
@@ -1536,6 +1536,17 @@ iiif:
...
+            subdomains:
+                - "{instance}--cdn-iiif"
+            healthcheck:
+                path: /ping-fastly
+                check-interval: 30000
+                timeout: 2000
+            surrogate-keys:
+                article-id-version:
+                    url: "^/lax:([0-9]+)/elife-([0-9]+)-([a-z0-9]+)-v([0-9]+)\\.(.+)$"
+                    value: "article/\\1 article/\\1/version/\\4"
```
Comment (thewilkybarkid) [inline projects/elife.yaml]: `article/1234v1` or `article/1234/v1` maybe?

File: projects/elife.yaml (hunk 1 above)
Comment (thewilkybarkid) [inline projects/elife.yaml]: The `-download` files aren't, as they're stream through Journal instead.

File: src/buildercore/cfngen.py [hunk 3]
```diff
@@ -260,7 +260,7 @@ def _parameterize_hostname(b):
         b['hostname'] = parameterize(b['hostname'])
         return b
 
-    if 'fastly' in context['project']['aws']:
+    if 'fastly' in context['project']['aws'] and context['project']['aws']['fastly']:
```
Comment (thewilkybarkid) [inline src/buildercore/cfngen.py]: Could be simplified with a `.get()`?

Comment (giorgiosironi) [reply src/buildercore/cfngen.py]: Yes, it should

**Comment by giorgiosironi** [comment]
Applying everywhere while the build runs.

**Comment by giorgiosironi** [comment]
And, purging all from `generic-cdn` and `iiif` to allow for the new surrogate keys to be used.

**Pull request merged**